Sources/NIO/ByteBuffer-aux.swift

```diff
@@ -373,6 +373,34 @@ extension ByteBuffer {
         self._moveWriterIndex(forwardBy: written)
         return written
     }
+    
+    /// Writes `byte` `repeated` times. Moves the writer index forward by the number of bytes written.
+    ///
+    /// - parameter byte: The `UInt8` byte to repeat.
+    /// - parameter count: How many times to repeat the given `byte`
+    /// - returns: How many bytes were written.
+    @discardableResult
+    public mutating func writeRepeatingByte(_ byte: UInt8, count: Int) -> Int {
+        precondition(count >= 0, "Can't write less than 0 bytes")
+        return self.writeWithUnsafeMutableBytes(minimumWritableBytes: count) { (pointer) -> Int in
+            memset(pointer.baseAddress!, Int32(byte), count)
+            return count
+        }
+    }
+    
+    /// Sets the given `byte` `repeated` times at the given `index`. Will reserve more memory if necessary. Does not move the writer index.
+    ///
+    /// - parameter byte: The `UInt8` byte to repeat.
+    /// - parameter count: How many times to repeat the given `byte`
+    /// - returns: How many bytes were written.
+    @discardableResult
+    public mutating func setRepeatingByte(_ byte: UInt8, count: Int, at index: Int) -> Int {
+        precondition(count >= 0, "Can't write less than 0 bytes")
```

Comment:
This method is missing a `isKnownUniquelyReferenced` check, so it doesn’t correctly maintain the value semantics of the type. Take a look at `_setBytes` for an example of what needs to be done here.

Reply:
The fact that this is passing its tests also means that there are no tests to confirm that this function actually does maintain the value semantics of the type: mind adding some?

Reply:
This thread is not outdated: it still has not been fixed.

Reply:
GH marked it as outdated automatically because there was another change.

Reply:
(When it's done I'll mark as resolved)